src/server/server.go

```diff
@@ -83,31 +92,32 @@ func (s *OAuthServer) startHTTP() {
 		if https {
 			bind := ":" + strconv.Itoa(s.config.HTTPSPort)
 			logger.Infof("Starting HTTPS server on [%v]", bind)
-			err := http.ListenAndServeTLS(bind, s.config.CertPublicKey, s.config.CertPrivateKey, r)
+
+			s.httpServer = &http.Server{
+				Addr:    bind,
+				Handler: r,
+			}
+
+			err := s.httpServer.ListenAndServeTLS(s.config.CertPublicKey, s.config.CertPrivateKey)
+
```

Comment:
You should keep the go routine concise, things that don’t need to be parallelized should be moved out of the go routine